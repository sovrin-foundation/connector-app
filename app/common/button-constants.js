// @flow
export const DENY = 'Deny'
export const CONNECT = 'Connect'
