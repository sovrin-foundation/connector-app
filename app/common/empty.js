// @flow
// sometimes we need an empty object to use as a placeholder in destructuring
export default {}
