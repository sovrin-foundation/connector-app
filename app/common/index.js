// @flow
export * from './button-constants'
export * from './message-constants'
export * from './route-constants'
export * from './secure-storage-constants'
export function noop() {}
