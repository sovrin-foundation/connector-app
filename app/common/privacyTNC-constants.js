// @flow

import { Platform } from 'react-native'

const isAndroid = Platform.OS === 'android'
export { EULA_URL as TermsAndConditionUrl } from '../eula/type-eula'
export const TermsAndConditionsTitle = 'Terms and Conditions'

export const PrivacyPolicyUrl = 'https://www.connect.me/privacy.html'
export const localPrivacyPolicySource = isAndroid
  ? 'file:///android_asset/external/connectme/privacy.html'
  : './privacy.html'
export const PrivacyPolicyTitle = 'Privacy Policy'
