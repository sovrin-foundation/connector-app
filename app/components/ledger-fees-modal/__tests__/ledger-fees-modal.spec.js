// @flow

import React from 'react'
import 'react-native'
import renderer from 'react-test-renderer'
import { STORE_STATUS } from '../../../common/type-common'
import type { GenericObject } from '../../../common/type-common'
import {
  LedgerFeesModalComponent,
  LedgerFeesModalStatus,
  LedgerFeesDescriptionText,
} from '../ledger-fees-modal'
import type { LedgerFeesModalProps } from '../ledger-fees-modal'
import { ERROR_GET_LEDGER_FEES } from '../../../store/ledger/type-ledger-store'
import type { LedgerFees } from '../../../store/ledger/type-ledger-store'

describe('<LedgerFeesModalComponent />', () => {
  const ledgerFees: LedgerFees = {
    data: {
      transfer: '0',
    },
    status: STORE_STATUS.IDLE,
    error: null,
  }

  function getCommonProps(
    props: LedgerFeesModalProps | GenericObject
  ): LedgerFeesModalProps {
    return {
      onYes: jest.fn(),
      onNo: jest.fn(),
      getLedgerFees: jest.fn(),
      resetLedgerFees: jest.fn(),
      isVisible: true,
      ledgerFees,
      walletBalance: '0',
      ...props,
    }
  }

  function setup(extraProps) {
    const props = getCommonProps(extraProps)
    const wrapper = renderer.create(<LedgerFeesModalComponent {...props} />)
    const instance = wrapper.root.findByType(LedgerFeesModalComponent).instance

    return { props, wrapper, instance }
  }

  it('trigger getLedgerFees to start fetching ledger fees', () => {
    const { wrapper, props } = setup({})
    expect(wrapper.toJSON()).toMatchSnapshot()
    expect(props.getLedgerFees).toHaveBeenCalled()
  })

  it('show loader, loading text, no buttons, if fetching fees', () => {
    const feesWithInProgress = {
      data: {
        transfer: '0',
      },
      status: STORE_STATUS.IN_PROGRESS,
      error: null,
    }
    const { wrapper } = setup({ ledgerFees: feesWithInProgress })
    expect(wrapper.toJSON()).toMatchSnapshot()
  })

  it('show error if fetching fees failed, show retry and cancel buttons', () => {
    const feesWithError = {
      data: {
        transfer: '0',
      },
      status: STORE_STATUS.ERROR,
      error: ERROR_GET_LEDGER_FEES('Test error'),
    }
    const { wrapper } = setup({ ledgerFees: feesWithError })
    expect(wrapper.toJSON()).toMatchSnapshot()
  })

  it('show transfer fees, Yes & No buttons, if fees is fetched', () => {
    const fees = {
      data: {
        transfer: '0.01',
      },
      status: STORE_STATUS.SUCCESS,
      error: null,
    }
    const { wrapper } = setup({ ledgerFees: fees, walletBalance: '1' })
    expect(wrapper.toJSON()).toMatchSnapshot()
  })

  it('show text passed by consumer of component, if renderFeesText is passed', () => {
    const fees = {
      data: {
        transfer: '0.01',
      },
      status: STORE_STATUS.SUCCESS,
      error: null,
    }
    const renderFeesText = (fees: string, status: string) => {
      switch (status) {
        case LedgerFeesModalStatus.TRANSFER_POSSIBLE_WITH_FEES:
          return (
            <LedgerFeesDescriptionText>
              Passed message {fees}
            </LedgerFeesDescriptionText>
          )
        default:
          return null
      }
    }
    const { wrapper } = setup({
      ledgerFees: fees,
      walletBalance: '1',
      renderFeesText,
    })
    expect(wrapper.toJSON()).toMatchSnapshot()
  })
})
