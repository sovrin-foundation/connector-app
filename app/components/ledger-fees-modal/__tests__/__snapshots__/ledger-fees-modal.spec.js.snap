// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LedgerFeesModalComponent /> show error if fetching fees failed, show retry and cancel buttons 1`] = `
<View
  fifth={true}
  shadow={true}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "elevation": 10,
      "marginHorizontal": 30,
      "shadowColor": "rgba(0, 0, 0, 0.2)",
      "shadowOffset": Object {
        "height": 10,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 30,
    }
  }
>
  <View
    spaceBetween={true}
    style={
      Object {
        "borderBottomColor": "#3f3f3f",
        "borderBottomWidth": 0.2,
        "justifyContent": "space-between",
        "padding": 20,
      }
    }
  >
    <View
      accessibilityLabel="ledger-fees-modal-header-icon"
      accessible={true}
      center={true}
      clearBg={true}
      iconStyle={
        Array [
          Object {
            "margin": 10,
          },
        ]
      }
      medium={true}
      resizeMode="contain"
      src={1}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "justifyContent": "center",
        }
      }
      testID="ledger-fees-modal-header-icon"
    >
      <View
        style={Object {}}
      >
        <Image
          accessibilityLabel="ledger-fees-modal-header-icon"
          accessible={true}
          resizeMode="contain"
          source={1}
          style={
            Array [
              Object {
                "height": 30,
                "width": 30,
              },
              null,
              Object {
                "margin": 10,
              },
            ]
          }
          testID="ledger-fees-modal-header-icon"
        />
      </View>
    </View>
    <Text
      accessibilityLabel="ledger-fees-modal-content"
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 17,
          },
          Object {
            "fontFamily": "Lato",
          },
          Object {
            "backgroundColor": "#ffffff",
            "color": "#777777",
          },
          null,
          Object {
            "textAlign": "center",
          },
          Object {
            "backgroundColor": "transparent",
          },
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          Object {
            "marginBottom": 5,
          },
        ]
      }
      testID="ledger-fees-modal-content"
    >
      There was a problem checking transaction fees. Retry?
    </Text>
  </View>
  <View
    row={true}
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "marginLeft": 15,
              "marginRight": 15,
            },
            undefined,
            Object {
              "marginLeft": 0,
              "marginRight": 0,
            },
            undefined,
          ]
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel="ledger-fees-modal-Cancel"
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ]
          }
          testID="ledger-fees-modal-Cancel"
          tvParallaxProperties={undefined}
        >
          <View
            pointerEvents="box-only"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#9E9E9E",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 19,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Object {
                  "backgroundColor": "#ffffff",
                },
                undefined,
                Object {
                  "padding": 12,
                },
                undefined,
                undefined,
                undefined,
                Array [
                  null,
                ],
                undefined,
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 16,
                    },
                    Object {
                      "color": "#4A8FE3",
                    },
                    Object {
                      "fontSize": 14,
                    },
                    undefined,
                    Object {
                      "fontWeight": "bold",
                    },
                    undefined,
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              Cancel
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "marginLeft": 15,
              "marginRight": 15,
            },
            undefined,
            Object {
              "marginLeft": 0,
              "marginRight": 0,
            },
            undefined,
          ]
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel="ledger-fees-modal-Retry"
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ]
          }
          testID="ledger-fees-modal-Retry"
          tvParallaxProperties={undefined}
        >
          <View
            pointerEvents="box-only"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#9E9E9E",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 19,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Object {
                  "backgroundColor": "#ffffff",
                },
                undefined,
                Object {
                  "padding": 12,
                },
                undefined,
                undefined,
                undefined,
                Array [
                  null,
                ],
                undefined,
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 16,
                    },
                    Object {
                      "color": "#4A8FE3",
                    },
                    Object {
                      "fontSize": 14,
                    },
                    undefined,
                    Object {
                      "fontWeight": "bold",
                    },
                    undefined,
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              Retry
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`<LedgerFeesModalComponent /> show loader, loading text, no buttons, if fetching fees 1`] = `
<View
  fifth={true}
  shadow={true}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "elevation": 10,
      "marginHorizontal": 30,
      "shadowColor": "rgba(0, 0, 0, 0.2)",
      "shadowOffset": Object {
        "height": 10,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 30,
    }
  }
>
  <View
    spaceBetween={true}
    style={
      Object {
        "borderBottomColor": "#3f3f3f",
        "borderBottomWidth": 0.2,
        "justifyContent": "space-between",
        "padding": 20,
      }
    }
  >
    <View
      center={true}
      style={
        Object {
          "alignItems": "center",
          "height": 40,
          "justifyContent": "center",
        }
      }
    >
      <View
        center={true}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "padding": 20,
          }
        }
      >
        <Image
          collapsable={undefined}
          source={1}
          style={
            Object {
              "height": 30,
              "tintColor": "#777777",
              "transform": Array [
                Object {
                  "rotate": "0deg",
                },
              ],
              "width": 30,
            }
          }
        />
      </View>
    </View>
    <Text
      accessibilityLabel="ledger-fees-modal-content"
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 17,
          },
          Object {
            "fontFamily": "Lato",
          },
          Object {
            "backgroundColor": "#ffffff",
            "color": "#777777",
          },
          null,
          Object {
            "textAlign": "center",
          },
          Object {
            "backgroundColor": "transparent",
          },
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          Object {
            "marginBottom": 5,
          },
        ]
      }
      testID="ledger-fees-modal-content"
    >
      Getting fees...
    </Text>
  </View>
</View>
`;

exports[`<LedgerFeesModalComponent /> show text passed by consumer of component, if renderFeesText is passed 1`] = `
<View
  fifth={true}
  shadow={true}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "elevation": 10,
      "marginHorizontal": 30,
      "shadowColor": "rgba(0, 0, 0, 0.2)",
      "shadowOffset": Object {
        "height": 10,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 30,
    }
  }
>
  <View
    spaceBetween={true}
    style={
      Object {
        "borderBottomColor": "#3f3f3f",
        "borderBottomWidth": 0.2,
        "justifyContent": "space-between",
        "padding": 20,
      }
    }
  >
    <View
      accessibilityLabel="ledger-fees-modal-header-icon"
      accessible={true}
      center={true}
      clearBg={true}
      iconStyle={
        Array [
          Object {
            "margin": 10,
          },
        ]
      }
      medium={true}
      resizeMode="contain"
      src={1}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "justifyContent": "center",
        }
      }
      testID="ledger-fees-modal-header-icon"
    >
      <View
        style={Object {}}
      >
        <Image
          accessibilityLabel="ledger-fees-modal-header-icon"
          accessible={true}
          resizeMode="contain"
          source={1}
          style={
            Array [
              Object {
                "height": 30,
                "width": 30,
              },
              null,
              Object {
                "margin": 10,
              },
            ]
          }
          testID="ledger-fees-modal-header-icon"
        />
      </View>
    </View>
    <Text
      accessibilityLabel="ledger-fees-modal-title"
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 40,
          },
          Object {
            "fontFamily": "Lato",
          },
          Object {
            "backgroundColor": "#ffffff",
            "color": "#777777",
          },
          Object {
            "fontWeight": "bold",
          },
          Object {
            "textAlign": "center",
          },
          Object {
            "backgroundColor": "transparent",
          },
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          Object {
            "marginBottom": 5,
          },
        ]
      }
      testID="ledger-fees-modal-title"
    >
      0.01
    </Text>
    <Text
      accessibilityLabel="ledger-fees-modal-title"
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 20,
          },
          Object {
            "fontFamily": "Lato",
          },
          Object {
            "backgroundColor": "#ffffff",
            "color": "#777777",
          },
          Object {
            "fontWeight": "bold",
          },
          Object {
            "textAlign": "center",
          },
          Object {
            "backgroundColor": "transparent",
          },
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          Object {
            "marginBottom": 5,
          },
        ]
      }
      testID="ledger-fees-modal-title"
    >
      Transaction fee
    </Text>
    <Text
      accessibilityLabel="ledger-fees-modal-content"
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 17,
          },
          Object {
            "fontFamily": "Lato",
          },
          Object {
            "backgroundColor": "#ffffff",
            "color": "#777777",
          },
          null,
          Object {
            "textAlign": "center",
          },
          Object {
            "backgroundColor": "transparent",
          },
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          Object {
            "marginBottom": 5,
          },
        ]
      }
      testID="ledger-fees-modal-content"
    >
      Passed message 
      0.01
    </Text>
  </View>
  <View
    row={true}
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "marginLeft": 15,
              "marginRight": 15,
            },
            undefined,
            Object {
              "marginLeft": 0,
              "marginRight": 0,
            },
            undefined,
          ]
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel="ledger-fees-modal-No"
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ]
          }
          testID="ledger-fees-modal-No"
          tvParallaxProperties={undefined}
        >
          <View
            pointerEvents="box-only"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#9E9E9E",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 19,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Object {
                  "backgroundColor": "#ffffff",
                },
                undefined,
                Object {
                  "padding": 12,
                },
                undefined,
                undefined,
                undefined,
                Array [
                  null,
                ],
                undefined,
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 16,
                    },
                    Object {
                      "color": "#4A8FE3",
                    },
                    Object {
                      "fontSize": 14,
                    },
                    undefined,
                    Object {
                      "fontWeight": "bold",
                    },
                    undefined,
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              No
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "marginLeft": 15,
              "marginRight": 15,
            },
            undefined,
            Object {
              "marginLeft": 0,
              "marginRight": 0,
            },
            undefined,
          ]
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel="ledger-fees-modal-Yes"
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ]
          }
          testID="ledger-fees-modal-Yes"
          tvParallaxProperties={undefined}
        >
          <View
            pointerEvents="box-only"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#9E9E9E",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 19,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Object {
                  "backgroundColor": "#ffffff",
                },
                undefined,
                Object {
                  "padding": 12,
                },
                undefined,
                undefined,
                undefined,
                Array [
                  null,
                ],
                undefined,
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 16,
                    },
                    Object {
                      "color": "#4A8FE3",
                    },
                    Object {
                      "fontSize": 14,
                    },
                    undefined,
                    Object {
                      "fontWeight": "bold",
                    },
                    undefined,
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              Yes
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`<LedgerFeesModalComponent /> show transfer fees, Yes & No buttons, if fees is fetched 1`] = `
<View
  fifth={true}
  shadow={true}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "elevation": 10,
      "marginHorizontal": 30,
      "shadowColor": "rgba(0, 0, 0, 0.2)",
      "shadowOffset": Object {
        "height": 10,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 30,
    }
  }
>
  <View
    spaceBetween={true}
    style={
      Object {
        "borderBottomColor": "#3f3f3f",
        "borderBottomWidth": 0.2,
        "justifyContent": "space-between",
        "padding": 20,
      }
    }
  >
    <View
      accessibilityLabel="ledger-fees-modal-header-icon"
      accessible={true}
      center={true}
      clearBg={true}
      iconStyle={
        Array [
          Object {
            "margin": 10,
          },
        ]
      }
      medium={true}
      resizeMode="contain"
      src={1}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "justifyContent": "center",
        }
      }
      testID="ledger-fees-modal-header-icon"
    >
      <View
        style={Object {}}
      >
        <Image
          accessibilityLabel="ledger-fees-modal-header-icon"
          accessible={true}
          resizeMode="contain"
          source={1}
          style={
            Array [
              Object {
                "height": 30,
                "width": 30,
              },
              null,
              Object {
                "margin": 10,
              },
            ]
          }
          testID="ledger-fees-modal-header-icon"
        />
      </View>
    </View>
    <Text
      accessibilityLabel="ledger-fees-modal-title"
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 40,
          },
          Object {
            "fontFamily": "Lato",
          },
          Object {
            "backgroundColor": "#ffffff",
            "color": "#777777",
          },
          Object {
            "fontWeight": "bold",
          },
          Object {
            "textAlign": "center",
          },
          Object {
            "backgroundColor": "transparent",
          },
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          Object {
            "marginBottom": 5,
          },
        ]
      }
      testID="ledger-fees-modal-title"
    >
      0.01
    </Text>
    <Text
      accessibilityLabel="ledger-fees-modal-title"
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 20,
          },
          Object {
            "fontFamily": "Lato",
          },
          Object {
            "backgroundColor": "#ffffff",
            "color": "#777777",
          },
          Object {
            "fontWeight": "bold",
          },
          Object {
            "textAlign": "center",
          },
          Object {
            "backgroundColor": "transparent",
          },
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          Object {
            "marginBottom": 5,
          },
        ]
      }
      testID="ledger-fees-modal-title"
    >
      Transaction fee
    </Text>
    <Text
      accessibilityLabel="ledger-fees-modal-content"
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 17,
          },
          Object {
            "fontFamily": "Lato",
          },
          Object {
            "backgroundColor": "#ffffff",
            "color": "#777777",
          },
          null,
          Object {
            "textAlign": "center",
          },
          Object {
            "backgroundColor": "transparent",
          },
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          Object {
            "marginBottom": 5,
          },
        ]
      }
      testID="ledger-fees-modal-content"
    >
      This transaction will cost you
       
      <Text
        accessibilityLabel="ledger-fees-modal-content"
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "Lato",
            },
            Object {
              "backgroundColor": "#ffffff",
              "color": "#777777",
            },
            Object {
              "fontWeight": "bold",
            },
            Object {
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "marginBottom": 5,
            },
          ]
        }
        testID="ledger-fees-modal-content"
      >
        0.01
      </Text>
       
      Sovrin Tokens. Do you wish to continue?
    </Text>
  </View>
  <View
    row={true}
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "marginLeft": 15,
              "marginRight": 15,
            },
            undefined,
            Object {
              "marginLeft": 0,
              "marginRight": 0,
            },
            undefined,
          ]
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel="ledger-fees-modal-No"
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ]
          }
          testID="ledger-fees-modal-No"
          tvParallaxProperties={undefined}
        >
          <View
            pointerEvents="box-only"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#9E9E9E",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 19,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Object {
                  "backgroundColor": "#ffffff",
                },
                undefined,
                Object {
                  "padding": 12,
                },
                undefined,
                undefined,
                undefined,
                Array [
                  null,
                ],
                undefined,
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 16,
                    },
                    Object {
                      "color": "#4A8FE3",
                    },
                    Object {
                      "fontSize": 14,
                    },
                    undefined,
                    Object {
                      "fontWeight": "bold",
                    },
                    undefined,
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              No
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "marginLeft": 15,
              "marginRight": 15,
            },
            undefined,
            Object {
              "marginLeft": 0,
              "marginRight": 0,
            },
            undefined,
          ]
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel="ledger-fees-modal-Yes"
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ]
          }
          testID="ledger-fees-modal-Yes"
          tvParallaxProperties={undefined}
        >
          <View
            pointerEvents="box-only"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#9E9E9E",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 19,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Object {
                  "backgroundColor": "#ffffff",
                },
                undefined,
                Object {
                  "padding": 12,
                },
                undefined,
                undefined,
                undefined,
                Array [
                  null,
                ],
                undefined,
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 16,
                    },
                    Object {
                      "color": "#4A8FE3",
                    },
                    Object {
                      "fontSize": 14,
                    },
                    undefined,
                    Object {
                      "fontWeight": "bold",
                    },
                    undefined,
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              Yes
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`<LedgerFeesModalComponent /> trigger getLedgerFees to start fetching ledger fees 1`] = `
<View
  fifth={true}
  shadow={true}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "elevation": 10,
      "marginHorizontal": 30,
      "shadowColor": "rgba(0, 0, 0, 0.2)",
      "shadowOffset": Object {
        "height": 10,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 30,
    }
  }
>
  <View
    spaceBetween={true}
    style={
      Object {
        "borderBottomColor": "#3f3f3f",
        "borderBottomWidth": 0.2,
        "justifyContent": "space-between",
        "padding": 20,
      }
    }
  >
    <View
      center={true}
      style={
        Object {
          "alignItems": "center",
          "height": 40,
          "justifyContent": "center",
        }
      }
    >
      <View
        center={true}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "padding": 20,
          }
        }
      >
        <Image
          collapsable={undefined}
          source={1}
          style={
            Object {
              "height": 30,
              "tintColor": "#777777",
              "transform": Array [
                Object {
                  "rotate": "0deg",
                },
              ],
              "width": 30,
            }
          }
        />
      </View>
    </View>
    <Text
      accessibilityLabel="ledger-fees-modal-content"
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 17,
          },
          Object {
            "fontFamily": "Lato",
          },
          Object {
            "backgroundColor": "#ffffff",
            "color": "#777777",
          },
          null,
          Object {
            "textAlign": "center",
          },
          Object {
            "backgroundColor": "transparent",
          },
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          Object {
            "marginBottom": 5,
          },
        ]
      }
      testID="ledger-fees-modal-content"
    >
      Getting fees...
    </Text>
  </View>
</View>
`;
